### DISCUSSION TITLE
Basic setup and use

### DISCUSSION THREAD

**Discussion by dusherwo57** [bgregs514/vscode-tm1#22] [Q&A] [answered]
I'm having a look at vscode-tm1 and have managed to install it into Visual Studio Code. I'm not clear what happens next. Do you open .PRO and .RUX files directly from the server folder? If so then you would need to restart to apply the changes. Or is there a Rest API link (but I've not spotted one so far). Think some basic steps would be helpful.

**Comment by EddStuart** [comment, marked as answer]
Hi David, hope you are well
I did spend a little time working this out, these are the steps I followed:
- Downloaded the project to local disk and unzipped
- In the terminal, I navigated to the project and ran the command "code --install-extension vscode-tm1-0.0.1.vsix"
- I clicked on the new cube icon in the left-hand menu bar (underneath Extensions), I think I closed and re-opened Visual Studio for this to appear

![image](https://user-images.githubusercontent.com/16777953/112154237-fa65b400-8bdb-11eb-8e27-32968b758da7.png)

- Select the cube to see "TM1 Connections" and a button for "Add New Connection" 

![image](https://user-images.githubusercontent.com/16777953/112154294-08b3d000-8bdc-11eb-951a-599357e834e8.png)

- Select Add New Connection to be taken to the setting page for "vscode-tm1.defaultConnection"
- Input the details for the Adminhost/ TM1Server dependant upon IntegratedSecurityMode
- If successful then you should see a treeview of Cubes, Processes, Control Cubes, Control Processes and Bookmarks

![image](https://user-images.githubusercontent.com/16777953/112153966-becaea00-8bdb-11eb-9440-5873cbc53216.png)

**Reply by bgregs514** [reply]
Thank you @EddStuart!  This is the correct answer.  Behind the scenes vscode-tm1 will check for the local workspace (determined by the configuration/settings.json page) and download local copies of the instance files.

One thing to note, the current "TM1 Connections" tab isn't fully functional (sort of a placeholder at this time), but the "refresh" button will reprocess the settings.  If you need to update your settings, clicking this refresh button will allow any new changes to take effect without needing to restart.

**Reply by dusherwo57** [reply]
Thanks! I was successful using your guidance, however in my environment there was no 'Refresh' button - but (after googling) Ctrl-R worked.
I do think you need a more detailed 'getting started' doc. I will do one for my colleagues and if it measures up I'm happy to pass it over.

**Comment by bgregs514** [comment]
Thanks for checking out  the project @dusherwo57!  Documentation is still a work in progress, but I agree, we should prioritize this now that the project is gaining more attention.

Much of the code is still changing at a pretty high frequency, and with that, the workflow as well.  I believe we are very close to finalizing the workflow plans, which means that we can start to document the process without worrying about it changing.  Thanks for your response, and I'll add this to our current backlog.

**Comment by patrunopablo** [comment]
HI,

I am trying to work with this extension but when I clic on add new connection nothing happens.

How can I add a new connection?

Thanks
